src/hll.c

```diff
@@ -2824,12 +2826,21 @@ hll_set_max_sparse(PG_FUNCTION_ARGS)
 {
     int32 old_maxsparse = g_max_sparse;
     int32 maxsparse = PG_GETARG_INT32(0);
+    StringInfoData sql;
 
     if (maxsparse < -1)
         ereport(ERROR,
                 (errcode(ERRCODE_DATA_EXCEPTION),
                  errmsg("sparse threshold must be in range [-1,MAXINT]")));
 
+    if (IS_QUERY_DISPATCHER())
```

Comment:
I remember @kainwen said `IS_QUERY_DISPATCHER` does not work well when `SingleQE`.

but I think `if (GpIdentity.segindex == MASTER_CONTENT_ID)` is really what we want.

does he check this?

Reply:
We need to check `Gp_role` is `GP_ROLE_DISPATCH`, because only QD can do dispatch.